### ISSUE TITLE
Weird behaviour / not working with a proxy

### ISSUE THREAD

**Issue by jbackes** [jfrog/jfrog-vscode-extension#97]
**Describe the bug**
We have an X-Ray server only reachable over a proxy server; lets say X-Ray server is https://xray and proxy is http://proxy:6789; note that proxy is http and X-Ray is https. I set the proxy in vscode and this seems to be working perfectly fine, i.e. I can install plugins, update vscode and wireshark for example logs from my ip to proxy ip: CONNECT update.code.visualstudio.com:443 HTTP/1.1 which succeeds with "Connection Established" etc.
But after entering the X-Ray configuration, the status bar just says "Checking connection with Xray server..." for about maybe 10 minutes and then an alert window pops up saying "Socket hang up".
Wireshark logs something weird (at least in my opinion): There is no CONNECT to the proxy but an http POST to the proxy with request url https://xray/api/v1/summary/component. In my opinion, wireshark should not see this POST because the connection should be ssl/tls encrypted but it seems like the plugin never tries to establish an ssl connection.

**To Reproduce**
Try to configure an https xray over an http proxy

**Expected behavior**
A successful connection to xray

**Versions**
- JFrog VS-Code extension version: 1.6.0
- JFrog VS-Code extension operating system: Windows 10 19042.746
- JFrog Xray Version: 2.7.6

**Comment by PiyushJonnie** [comment]
Steps to reproduce the issue :

- Install latest VSCode
- Install latest version of JFrog VSCode plugin
- Connect the plugin to Artifactory
- Enter JFrog platform URL
- Accept proposed Artifactory URL
- Accept proposed Xray URL
- Enter JFrog access token
- Wait for the connection to establish

Visual Studio Code error message window appears with text: “Request failed with status code 400”

**Comment by eyalbe4** [comment]
Thank you for reporting this issue @PiyushJonnie!
Is you JFrog environment accessible through an HTTP proxy?
If so, ddi you set your proxy details as described [here](https://marketplace.visualstudio.com/items?itemName=JFrog.jfrog-vscode-extension#proxy-configuration)?

**Comment by PiyushJonnie** [comment]
@eyalbe4 Yes, here JFrog environment is only accessible through an HTTP proxy. Proxy settings are configured as mentioned in the link. If the proxy is disabled, the response changes to:

“Could not connect to Xray: Error connect ETIMEDOUT <ip address>:443”

Other extensions are working correctly and are able to use the proxy as expected.

**Comment by jghal** [comment]
And just to confirm the negative, I am able to connect the extension to this same [public-facing] Artifactory server when off the corporate network and not using a proxy.

**Comment by jghal** [comment]
Additionally. Both `jfrog` and `frogbot` can connect to the Artifactory server with the proxy with `http[s]_proxy` env variables set. I've tried with and without proxy authentication configured, though I don't actually need to authenticate the proxy for this server (confirmed with `jfrog` and `curl`). When I set the extension log level to debug, I don't see any extra info in the extension logs versus info logging.

**Comment by attiasas** [comment]
Our apologies for the inconvenience caused by this.
This PR - [jfrog-client-js/pull/73](https://github.com/jfrog/jfrog-client-js/pull/73) should fix it.
Fix released from version [2.2.0](https://github.com/jfrog/jfrog-vscode-extension/releases/tag/2.2.0)+
